spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -494,6 +506,20 @@ public static boolean extensionsEnabled(SparkSession spark) {
     return extensions.contains("IcebergSparkSessionExtensions");
   }
 
+  public static boolean isPartitioned(SparkSession spark, Path tableLocation) {
+    try {
+      return Arrays.stream(
+              FileSystem.get(spark.sparkContext().hadoopConfiguration()).listStatus(tableLocation))
```

Comment:
checkstyle favor `spark.sessionState().newHadoopConf()` over ` spark.sparkContext().hadoopConfiguration()` as session config is not lost